src/rqt_topic/topic_info.py

```diff
@@ -86,7 +85,7 @@ def stop_monitoring(self):
         self.monitoring = False
         self._reset_data()
         if self._subscriber is not None:
-            self._spinner.register_listeners_for_destruction(self._subscriber)
+            self._node.destroy_subscription(self._subscriber)
```

Comment:
```suggestion
            # Note (mlautman): This segfaults because of an underlying bug in rclpy.
            #                              See: https://github.com/ros2/rclpy/issues/255 
            self._node.destroy_subscription(self._subscriber)
```